### DISCUSSION TITLE
Issues after upgrading to v4.20c

### DISCUSSION THREAD

**Discussion by sagamusix** [AFLplusplus/AFLplusplus#2068] [Q&A] [answered]
I recently upgraded from v4.10c to v4.20c, and now I'm seeing lots of unexpected crashes being collected in all sorts of random places. They're all segmentation violations trying to access null pointers as far as I can see. This did not happen at all with v4.10c, fuzzing the exact same code (actually across several different versions of the software that I'm fuzzing simultaneously), so I believe something must have changed in v4.20c that doesn't work properly with my setup.

Here's all the information I can think of that might be relevant:
- I building and running on Linux using LLVM 16 and using afl-clang-lto++
- I build several versions of my code with different afl options: Some with AFL_HARDEN=1 and the others with AFL_USE_ASAN=1 - they all have the same issue
- The new PERFORMANCE=1 is not being used
- Re-running the binary through afl-showmap with v4.20c reproduces the crash, recompiling it with v4.10c it's also gone.

Any ideas what could be causing this?

The code in question is https://github.com/OpenMPT/openmpt (fuzzing setup can be found in contrib/fuzzing)

**Comment by vanhauser-thc** [comment, marked as answer]
can you try the current dev branch state please, there was a fix for LTO?
if this problem persists please give more information because there is not much I can work with. it works fine for me, on our CI and nobody else is reporting issues.

**Reply by sagamusix** [reply]
Thanks, the problem is indeed gone with the latest dev version!
